discovery-infra/tests/base_test.py

```diff
@@ -25,13 +22,21 @@ def random_name():
 class BaseTest:
 
     @pytest.fixture(scope="function")
-    def nodes(self, setup_node_controller):
-        controller = setup_node_controller
-        nodes = Nodes(controller, env_variables["private_ssh_key_path"])
-        nodes.set_correct_boot_order(start_nodes=False)
-        yield nodes
-        nodes.shutdown_all()
-        nodes.format_all_disks()
+    def nodes(self, setup_node_controller, is_dev_env):
+        net_asset = None
+        try:
+            if is_dev_env:
+                net_asset = NetworkAssets()
+                env_variables["net_asset"] = net_asset.get()
+            controller = setup_node_controller(**env_variables)
+            controller.prepare_nodes()
```

Comment:
lets move it to nodes.prepare_nodes()

Reply:
not in this scope. it requires more tests on both envs. 
why do we want that? isnt it OK that its done from the controller?

Reply:
Fixed